### PR TITLE
Added new curriculum download UI section

### PR DESCRIPTION
**Pull request by orangemug**
## Description
Added download UI section for curriculum download. Also added "Back to KS4 options" button into the UI.

## Issue(s)
Fixes `CUR-778`

## How to test

1. Go to {owa_deployment_url}/teachers/curriculum/science-secondary-edexcel/downloads
2. Check for "Back to KS4 options" button
3. Check UI implemented as expected

## Screenshots

Screenshot 2024-08-08 at 16 25 04 Screenshot 2024-08-08 at 16 25 10

### CODE REVIEW

File: src/components/CurriculumComponents/CurriculumDownloadTab/index.tsx [hunk 1]
```diff
@@ -233,6 +233,13 @@ const CurriculumDownloadTab: FC<CurriculumDownloadTabProps> = ({
     );
   }
 
+  let onBackToK4sOptions: undefined | (() => void);
```
Comment (rightsaidjames) [inline src/components/CurriculumComponents/CurriculumDownloadTab/index.tsx:236]: Typo - should be `Ks4Options`

File: src/components/CurriculumComponents/CurriculumDownloadView/index.tsx [hunk 2]
```diff
@@ -126,6 +142,18 @@ const CurriculumDownloadView: FC<CurriculumDownloadViewProps> = ({
...
         Download
       </OakHeading>
+      {onBackToK4sOptions && (
+        <Box $mb={24}>
+          <Button
+            variant={"buttonStyledAsLink"}
+            icon="chevron-left"
+            data-testid="back-to-downloads-link"
+            size="small"
+            label="Back to K4S options"
```
Comment (rightsaidjames) [inline src/components/CurriculumComponents/CurriculumDownloadView/index.tsx:152]: Typo in the label as well

**Comment by rightsaidjames** [comment]
@orangemug Can you show a screenshot of the desktop view for the new downloads UI, so I can see it in two columns before this PR is merged?

File: src/components/CurriculumComponents/CurriculumDownloadView/index.tsx (hunk 2 above)
Comment (orangemug) [reply src/components/CurriculumComponents/CurriculumDownloadView/index.tsx:152]: Ta, will fix shortly

**Comment by orangemug** [comment]
Here's the two column layout @rightsaidjames 

Screenshot 2024-08-08 at 17 48 51

**Review by rightsaidjames — APPROVED** [review]
Looks good to me from a static analysis POV, obviously I'll test it more when it gets into the epic branch.

**Pull request merged**